Library/Homebrew/language/node.rb

```diff
@@ -4,6 +4,17 @@ def self.npm_cache_config
       "cache=#{HOMEBREW_CACHE}/npm_cache\n"
     end
 
+    def self.pack_for_installation
+      # npm 5.0.0 broke the assumption Homebrew made that npm would always
+      # create and install real directories when passed a file path. We can
+      # restore the old behaviour by packing the path first with "npm pack"
+      # which also helpfully spits out the filename it just created allowing
+      # us to pass that as the final argument.
```

Comment:
Basically this comment should answer the question: "why do we need to run `npm pack` at all?" I don't understand the answer to that despite having read most of the conversation and definitely not just from this code.